test/server/graphql/v2/mutation/MemberInvitationMutations.test.ts

```diff
@@ -0,0 +1,437 @@
+import { expect } from 'chai';
+import gqlV2 from 'fake-tag';
+import { describe, it } from 'mocha';
+import { createSandbox } from 'sinon';
+
+import ActivityTypes from '../../../../../server/constants/activities';
+import roles from '../../../../../server/constants/roles';
+import { idEncode, IDENTIFIER_TYPES } from '../../../../../server/graphql/v2/identifiers';
+import emailLib from '../../../../../server/lib/email';
+import models from '../../../../../server/models';
+import { fakeCollective, fakeUser } from '../../../../test-helpers/fake-data';
+import * as utils from '../../../../utils';
+
+let collectiveAdminUser, collective;
+let sandbox, sendEmailSpy;
+
+describe('MemberInvitationMutations', () => {
+  before(async () => {
+    await utils.resetTestDB();
+    sandbox = createSandbox();
+    sendEmailSpy = sandbox.spy(emailLib, 'sendMessage');
+    collectiveAdminUser = await fakeUser();
+    collective = await fakeCollective({ name: 'webpack test collective', admin: collectiveAdminUser });
+  });
+  after(() => sandbox.restore());
+
+  const inviteMemberMutation = gqlV2/* GraphQL */ `
+    mutation InviteMember(
+      $memberAccount: AccountReferenceInput!
+      $account: AccountReferenceInput!
+      $role: MemberRole!
+      $description: String
+      $since: DateTime
+    ) {
+      inviteMember(
+        memberAccount: $memberAccount
+        account: $account
+        role: $role
+        description: $description
+        since: $since
+      ) {
+        id
+        role
+        description
+        since
+      }
+    }
+  `;
+
+  describe('inviteMember', () => {
+    it('should create a new member invitation and its related activity', async () => {
+      sendEmailSpy.resetHistory();
+      const randomUserToInvite = await fakeUser();
+      const result = await utils.graphqlQueryV2(
+        inviteMemberMutation,
+        {
+          memberAccount: { id: idEncode(randomUserToInvite.collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          account: { id: idEncode(collective.id, IDENTIFIER_TYPES.ACCOUNT) },
```

Comment:
Here `randomUserToInvite` is created with id `4` and there seems to be a collective with type `ORGANIZATION` also present with `id: 4`. And since these are clashing `idEncode(randomUserToInvite.id, IDENTIFIER_TYPES.ACCOUNT)` is actually resolving to the collective with type `ORGANIZATION` and not the `User` one.

I don't know if this s=is limitation of `fakUser` or if it can be possible in real DB as well. So I have used `randomUserToInvite.collective.id` since `fakeUser` creates this with `type: USER`
https://github.com/opencollective/opencollective-api/blob/aaa0b0d03efe4b143f6967614a67d305862dfab0/test/test-helpers/fake-data.ts#L62-L65

Reply:
`randomUserToInvite` is a user (`Users`) table. `randomUserToInvite.collective` is its profile/account (`Collectives` table). They're two different things and thus it would make no sense to pass `memberAccount: user`.

Using `memberAccount: user.collective` is the right thing to do here.